File: 2021/Day01/solution.js
const fs = require('fs');
const data = fs.readFileSync('./input.txt', 'utf-8').split('\n').map(Number);

let part1 = 0;
for (let i = 1; i < data.length; i++) {
  if (data[i] > data[i-1]) part1++;
}

console.log(`Part 1: ${part1}`)

// Part 2
let part2 = 0;
let currentDepth = data[0]+data[1]+data[2];
for (let i = 3; i < data.length; i++) {
  const depth = data[i] + data[i-1] + data[i-2];

  if (depth > currentDepth) part2++;

  currentDepth = depth;
}

console.log(`Part 2: ${part2}`)

File: 2021/Day01/solution.py
data = [int(x) for x in open('./input.txt').read().splitlines()]


def check_for_increases(list):
    counter = 0
    for idx in range(len(list) - 1):
        if list[idx] < list[idx+1]:
            counter += 1

    return counter


part1 = check_for_increases(data)
print(f'Part 1: {part1}')

# Part 2
p2 = []
for i in range(len(data) - 2):
    p2.append(sum(data[i:i+3]))

part2 = check_for_increases(p2)
print(f'Part 2: {part2}')
